src/com/oltpbenchmark/LatencyRecord.java

```diff
@@ -1,170 +1,170 @@
-	static final int ALLOC_SIZE = 500000;
+    /**
+     * Allocate space for 500k samples at a time
+     */
+    static final int ALLOC_SIZE = 500000;
 
-	/**
-	 * Contains (start time, latency, transactionType, workerid, phaseid) pentiplets 
-	 * in microsecond form. The start times are "compressed" by encoding them as 
-	 * increments, starting from startNs. A 32-bit integer provides sufficient resolution
-	 * for an interval of 2146 seconds, or 35 minutes.
-	 */
-	private final ArrayList<Sample[]> values = new ArrayList<Sample[]>();
-	private int nextIndex;
+    /**
+     * Contains (start time, latency, transactionType, workerid, phaseid) pentiplets
+     * in microsecond form. The start times are "compressed" by encoding them as
+     * increments, starting from startNs. A 32-bit integer provides sufficient resolution
+     * for an interval of 2146 seconds, or 35 minutes.
+     */
+    private final ArrayList<Sample[]> values = new ArrayList<>();
+    private int nextIndex;
 
-	private final long startNs;
-	private long lastNs;
+    private final long startNs;
+    private long lastNs;
 
-	public LatencyRecord(long startNs) {
-		assert startNs > 0;
+    public LatencyRecord(long startNs) {
 
-		this.startNs = startNs;
-		lastNs = startNs;
-		allocateChunk();
 
-	}
+        this.startNs = startNs;
+        lastNs = startNs;
+        allocateChunk();
+
+    }
 
     public void addLatency(int transType, long startNs, long endNs, int workerId, int phaseId) {
-		assert lastNs > 0;
-		assert lastNs - 500 <= startNs;
-		assert endNs >= startNs;
-
-		if (nextIndex == ALLOC_SIZE) {
-			allocateChunk();
-		}
-		Sample[] chunk = values.get(values.size() - 1);
-
-		long startOffsetNs = (startNs - lastNs + 500);
-		assert startOffsetNs >= 0;
-		int latencyUs = (int) ((endNs - startNs + 500) / 1000);
-		assert latencyUs >= 0;
-
-		chunk[nextIndex] = new Sample(transType, startOffsetNs, latencyUs
-				, workerId, phaseId);
-		++nextIndex;
-
-		lastNs += startOffsetNs;
-	}
-
-	private void allocateChunk() {
-		assert (values.isEmpty() && nextIndex == 0)
-			|| nextIndex == ALLOC_SIZE;
-		values.add(new Sample[ALLOC_SIZE]);
-		nextIndex = 0;
-	}
-
-	/** Returns the number of recorded samples. */
-	public int size() {
-		// Samples stored in full chunks
-		int samples = (values.size() - 1) * ALLOC_SIZE;
-
-		// Samples stored in the last not full chunk
-		samples += nextIndex;
-		return samples;
-	}
-
-	/** Stores the start time and latency for a single sample. Immutable. */
-	public static final class Sample implements Comparable<Sample> {
-		public final int tranType;
-		public long startNs;
-		public final int latencyUs;
-		public final int workerId;
-		public final int phaseId;
+
+
+        if (nextIndex == ALLOC_SIZE) {
+            allocateChunk();
+        }
+        Sample[] chunk = values.get(values.size() - 1);
+
+        long startOffsetNs = (startNs - lastNs + 500);
+
+        int latencyUs = (int) ((endNs - startNs + 500) / 1000);
```

Comment:
TODO I wonder if this could cause integer division issues. Wonder why this was done?